### PR TITLE
dev/core#1452 Put text inside label tags for recurring contribution selection

### PR DESCRIPTION
**Pull request by larssandergreen**
Before: Part of text for recurring contributions selection on contribution page was outside label tags, making it impossible to style labels.

After: All text now inside label tags except for the "for" when offering instalments. This allows styling.

Once I got into this, it became clear that it was quite messy to do this as I intended. I'm not sure this is the best approach, so I'm putting up this pull request as is to see if folks have a better idea how to tackle this or support this ugly approach. I think I have maintained translatability here, but will need to test that further before merging. Will also see if I can get the "for" inside tags as well.

### CODE REVIEW

**Comment by larssandergreen** [comment]
jenkins, test this please

**Comment by demeritcowboy** [comment]
Related on the translation part: https://lab.civicrm.org/dev/translation/-/issues/53

**Comment by mlutfy** [comment]
Agree this is a very difficult one.

From a translation/string perspective, this PR does not seem to change the current behaviour, for better or for worse.

I'm not sure about the `frequency_interval` changes? Could you explain a bit more what is happening before/after? (screenshots?)

File: CRM/Contribute/Form/Contribution/Main.php [hunk 1]
```diff
@@ -752,15 +741,35 @@ public static function buildRecur(&$form) {
...
+    //$frequencyUnit->freeze( );
+    //}
+
+    $form->add('text', 'installments', ts('installments'),
+      $attributes['installments']
+    );
+    $form->addRule('installments', ts('Number of installments must be a whole number.'), 'integer');
+
+    $is_recur_label = ts('I want to contribute this amount every');
+    $unit_label = ts('Every');
```
Comment (seamuslee001) [inline CRM/Contribute/Form/Contribution/Main.php:756]: Where does this get used?

**Comment by larssandergreen** [comment]
@mlutfy I haven't changed anything about how frequency_interval works except its label. I did have to change the order of that block of code relative to others to get the variables needed in the right place.
What I did change was the frequency_interval.label, which was previously ts{'Every'}, but was not actually used in the template. What was used was frequency_interval directly. Now, I've set frequency_interval.label to $unit (which is month(s), day(s), etc) and use this as a label in the template. So the whole thing is just to move the same text inside a label, so it can be styled consistently. There is no change in functionality directly, but the HTML is different as follows:
Before:
image
After:
image

File: CRM/Contribute/Form/Contribution/Main.php (hunk 1 above)
Comment (larssandergreen) [reply CRM/Contribute/Form/Contribution/Main.php:756]: Good call, it does not get used and I've now removed it.

**Comment by larssandergreen** [comment]
@demeritcowboy I think this should play well with the code that @mlutfy has proposed to deal translation of months, days, etc.

**Comment by demeritcowboy** [comment]
I tried a couple variations. Seems ok.

For translation it seems no worse than before except now "I want to contribute this amount every" is a new string so that will be untranslated for possibly a few releases before the translations catch up, but Word Replacements works as a temp fix.

**Comment by mlutfy** [comment]
:+1: from me based on code review. I'd be comfortable to merge this based on @demeritcowboy 's review, but will leave open a bit in case Seamus or others have any comments.

**Comment by eileenmcnaughton** [comment]
8 days is long enough for merge-ready

**Pull request merged**